test/e2e/autoscale.go

```diff
@@ -16,6 +16,12 @@ limitations under the License.
 
 package e2e
 
+// Tests in this file can work with Chaosduck enabled for autoscaler.
+// These tests accept that the desired replicas could be dropped during
+// the test process. As Autoscaler is stateful, the new lead needs time
+// to collect scaling metrics, which could potentially result in scaling
+// down.
```

Comment:
This is supposed not to happen per the implementation, as the new scalers are supposed to start in panic mode. Do we have a bug?

Reply:
Yeah, I forgot about this, they shouldn't scale down.

Reply:
I was thinking that the same case from https://github.com/knative/serving/pull/9259#issuecomment-693175182 could happen if we enable Chaosduck. Now I think it is different here. In upgrade, the old and new autoscalers exists for a peroid and the activators send metrics to the old autoscaler while the new autoscaler has no metrics to scale. Here, the old lead pod has already been killed.

I think this test could be broken: https://github.com/knative/serving/blob/master/test/e2e/minscale_readiness_test.go#L126. If the lead pod is kept killed every 30 seconds, the old revision won't be scaled down within stable window. Maybe we can set the stable window to min value. I have created https://github.com/knative/serving/pull/9844. Let's see what happened.

Reply:
https://github.com/knative/serving/pull/9844 looks good. Closing this one.

Reply:
Yeah https://github.com/knative/serving/blob/master/test/e2e/minscale_readiness_test.go#L126. was broken in https://github.com/knative/serving/pull/9884